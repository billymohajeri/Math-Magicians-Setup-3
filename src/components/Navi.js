import React from 'react';
import { NavLink } from 'react-router-dom';

const Navi = () => (
  <nav className="aa">
    <h1>Math Magicians</h1>
    <ul>
      <li>
        <NavLink
          to="/"
          end
          className={({ isActive }) => (isActive ? 'activeLink' : undefined)}
        >
          Home
        </NavLink>
      </li>
      <li>
        <NavLink
          to="quote"
          className={({ isActive }) => (isActive ? 'activeLink' : undefined)}
        >
          Quote
        </NavLink>
      </li>
      <li>
        <NavLink
          to="calculator"
          className={({ isActive }) => (isActive ? 'activeLink' : undefined)}
        >
          Calculator
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default Navi;
