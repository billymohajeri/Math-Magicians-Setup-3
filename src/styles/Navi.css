.navmenu-desktop {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 2%;
  border-radius: 2px;
}

.nav-items {
  display: flex;
  align-items: center;
  margin-left: auto;
  list-style: none;
  margin-top: 0;
  margin-bottom: 0;
  color: wheat;
}

.nav-items li {
  margin: 1rem;
  font-size: 2rem;
}

.activeLink {
  color: #d47500;
}

.logo {
  color: wheat;
  font-size: 5rem;
  margin-top: 1%;
}
