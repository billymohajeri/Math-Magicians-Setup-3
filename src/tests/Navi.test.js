import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navi from '../components/Navi';

it('renders correctly', () => {
  const navi = renderer
    .create(
      <Router>
        <Navi />
      </Router>,
    )
    .toJSON();
  expect(navi).toMatchSnapshot();
});
