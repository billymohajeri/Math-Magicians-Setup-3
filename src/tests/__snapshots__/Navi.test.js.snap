// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navmenu-desktop"
>
  <h1
    className="logo"
  >
    Math Magicians
  </h1>
  <ul
    className="nav-items"
  >
    <li>
      <a
        aria-current="page"
        className="activeLink"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
    <li>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
  </ul>
</nav>
`;
